File: examples/university-demo/src/main/java/org/axonframework/examples/demo/university/faculty/write/subscribestudent/SubscribeStudentToCourse.java
package org.axonframework.examples.demo.university.faculty.write.subscribestudent;

import org.axonframework.examples.demo.university.shared.ids.CourseId;
import org.axonframework.examples.demo.university.shared.ids.StudentId;
import org.axonframework.modelling.annotation.TargetEntityId;

public record SubscribeStudentToCourse(StudentId studentId, CourseId courseId) {

    @TargetEntityId
    private SubscriptionId subscriptionId() {
        return new SubscriptionId(courseId, studentId);
    }
}

File: examples/university-demo-kotlin/src/main/kotlin/faculty/write/subscribe_student_fmodel/SubscribeStudentToCourse.kt
package org.axonframework.examples.university.faculty.write.subscribe_student_fmodel

import org.axonframework.examples.university.shared.ids.CourseId
import org.axonframework.examples.university.shared.ids.StudentId
import org.axonframework.examples.university.shared.ids.SubscriptionId
import org.axonframework.messaging.commandhandling.annotation.Command
import org.axonframework.modelling.annotation.TargetEntityId

data class SubscribeStudentToCourse(
    val studentId: StudentId,
    val courseId: CourseId,
) {
    @TargetEntityId
    val subscriptionId: SubscriptionId = SubscriptionId(studentId, courseId)
}
